tests/unittest_scoped_nodes.py

```diff
@@ -1040,7 +1040,7 @@ class Past(Present):
         astroid = builder.parse(data)
         past = astroid["Past"]
         attr = past.getattr("attr")
-        self.assertEqual(len(attr), 1)
+        self.assertEqual(len(attr), 1, attr)
```

Comment:
```suggestion
        assert len(attr) == 1
```
I think when using pytest this is possible (?)

Reply:
Any Ideas why this might have started failing though? I had a read through `getattr` but since it doesn't call `ancestors` directly I'm going to need to dig a little more.

Reply:
on `main`, `attr` is...
```
[<AssignName.attr l.3 at 0x[...]>]
```
with my changes this becomes...
```
[<AssignName.attr l.3 at 0x[...]>, <Const.int l.3 at 0x[...]>]
```

Reply:
Oh I was looking at the wrong `getattr` before... https://github.com/PyCQA/astroid/blob/f86d6e9f67e449209fe53a6e9cef4f7b55adf52a/astroid/nodes/scoped_nodes.py#L2514-L2517

Reply:
I suspect that using `ClassDef`s and `InferenceContext`s as keys was a bit naive: they aren't Singletons and don't implement `__hash__` and `__eq__` so I am only caching specific instances by object id instead of detecting when the `ancestor`s or `context`s are equivalent.